File: backend/ShiftMate.API/Models/Shift.cs
namespace ShiftMate.API.Models;

public class Shift
{
    public int Id { get; set; }

    public int LocationId { get; set; }

    public DateTime StartTime { get; set; }

    public DateTime EndTime { get; set; }

    public int RequiredEmployees { get; set; }

    public string? Notes { get; set; }

    public Location Location { get; set; } = null!;

    public ICollection<ShiftAssignment> ShiftAssignments { get; set; } = new List<ShiftAssignment>();

    public ICollection<ShiftRoleRequirement> RoleRequirements { get; set; } = new List<ShiftRoleRequirement>();
}

File: frontend/src/types/shift.ts
export interface Shift {
    id: number
    locationId: number
    locationName: string
    startTime: string
    endTime: string
    requiredEmployees: number
    notes: string | null
    status: number
}

export interface CreateShiftRequest {
    locationId: number
    startTime: string
    endTime: string
    requiredEmployees: number
    notes: string
}

export interface UpdateShiftStatusRequest {
    status: number
}